src/main/java/uk/gov/ons/ctp/response/casesvc/endpoint/CaseEndpoint.java

```diff
@@ -124,6 +131,51 @@ public CaseEndpoint(
     }
   }
 
+  @RequestMapping(method = RequestMethod.GET)
+  public ResponseEntity<List<CaseDetailsDTO>> findCases(
+      @RequestParam(required = false) String sampleUnitId,
+      @RequestParam(required = false) String partyId) {
+    log.info("Finding cases sampleUnitId={} partyId={}", sampleUnitId, partyId);
+    Example<Case> exampleCase = buildExampleCase(sampleUnitId, partyId);
+    List<Case> cases = getCases(exampleCase);
+    List<CaseDetailsDTO> caseResponses =
+        cases
+            .stream()
+            .map(
+                c -> {
+                  CaseDetailsDTO caseDetails = mapperFacade.map(c, CaseDetailsDTO.class);
+                  CaseGroup parentCaseGroup =
+                      caseGroupService.findCaseGroupByCaseGroupPK(c.getCaseGroupFK());
+                  caseDetails.setCaseGroup(mapperFacade.map(parentCaseGroup, CaseGroupDTO.class));
+                  return caseDetails;
+                })
+            .collect(Collectors.toList());
+
+    return ResponseEntity.ok(caseResponses);
+  }
+
+  private List<Case> getCases(Example<Case> exampleCase) {
+    // Limit to max 100 cases
+    PageRequest paging = new PageRequest(0, 100);
+    boolean emptyRequest = exampleCase.getProbe().equals(new Case());
+    if (emptyRequest) {
+      return caseRepository.findAll(paging).getContent();
```

Comment:
Is there any benefit to returning the first 100 cases if there is no way to get the rest?

Reply:
Possibly not was thinking otherwise it would be a select * for all cases which will probably be fine. I'll remove it

Reply:
It was mainly used as a LIMIT 100